### PR TITLE
[eas-cli] support removal of asc api keys

### PR DESCRIPTION
**Pull request by quinlanj**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.
- [x] I've tagged the changelog entry with `[EAS BUILD API]` if it's a part of a breaking change to EAS Build API (only possible when updating `@expo/eas-build-job` package).

# Why

Add support for removal of Asc Api Keys in the credentials manager

# Test Plan

- [ ] new tests pass

### CODE REVIEW

**Comment by quinlanj** [comment]
/changelog-entry new-feature Add support for removal of App Store Connect Api Keys.

File: packages/eas-cli/src/credentials/ios/actions/AscApiKeyUtils.ts [hunk 1]
```diff
@@ -163,3 +167,79 @@ async function getBestEffortIssuerIdAsync(
...
+}
+
+async function selectAscApiKeysAsync(
+  ascApiKeys: AppStoreConnectApiKeyFragment[]
+): Promise<AppStoreConnectApiKeyFragment | null> {
+  const sortedAscApiKeys = sortAscApiKeysByUpdatedAtDesc(ascApiKeys);
+  const { chosenAscApiKey } = await promptAsync({
+    type: 'select',
+    name: 'chosenAscApiKey',
+    message: 'Select an Api Key from the list.',
```
Comment (dsokal) [inline packages/eas-cli/src/credentials/ios/actions/AscApiKeyUtils.ts:203]:
```suggestion
    message: 'Select an Api Key from the list:',
```

File: packages/eas-cli/src/credentials/ios/actions/RemoveAscApiKey.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  }
+}
+
+export class RemoveAscApiKey {
+  constructor(private account: Account, private ascApiKey: AppStoreConnectApiKeyFragment) {}
+
+  public async runAsync(ctx: CredentialsContext): Promise<void> {
+    if (ctx.nonInteractive) {
+      throw new Error(`Cannot remove App Store Connect Api Keys in non-interactive mode`);
```
Comment (dsokal) [inline packages/eas-cli/src/credentials/ios/actions/RemoveAscApiKey.ts:26]:
```suggestion
      throw new Error(`Cannot remove App Store Connect Api Keys in non-interactive mode.`);
```

**Review by dsokal — APPROVED** [review]
🚀

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**